internal/app/osd/internal/reg/reg.go

```diff
@@ -397,3 +400,66 @@ func (r *Registrator) Top(ctx context.Context, in *empty.Empty) (reply *proto.To
 	reply = &proto.TopReply{ProcessList: p}
 	return
 }
+
+// DF implements the proto.OSDServer interface.
+func (r *Registrator) DF(ctx context.Context, in *empty.Empty) (reply *proto.DFReply, err error) {
+	var stat unix.Statfs_t
+	file, err := os.Open("/proc/mounts")
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	stats := []*proto.DFStat{}
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
```

Comment:
Do we want to filter out non filesystems?
```
if !strings.HasPrefix(scanner.Text(), "/") {
  continue
}
```

Reply:
I'm not sure TBH. I was trying to mimic what `df` does:

```
[I] $ df -h
Filesystem      Size  Used Avail Use% Mounted on
overlay          59G   39G   18G  70% /
tmpfs           2.0G     0  2.0G   0% /sys/fs/cgroup
osxfs           234G  114G  119G  50% /root
dev              10M     0   10M   0% /dev
shm              64M     0   64M   0% /dev/shm
/dev/sda1        59G   39G   18G  70% /run
```